OK: Create additional argument for the center in Centralization indexes

renanxcortes opened the pull request:
This is a PR that address https://github.com/pysal/segregation/issues/18.

I've been working in this issue and pushed this commit here with several approaches for all centralization indexes with this new argument:
```
    center        : string, tuple or integer.
                    This defines what is considered to be the center of the spatial context under study.

                    If string, this can be set to:
                        
                        "mean": the center longitude/latitude is the mean of longitudes/latitudes of all units. 
                        "median": the center longitude/latitude is the median of longitudes/latitudes of all units. 
                        "population_weighted_mean": the center longitude/latitude is the mean of longitudes/latitudes of all units weighted by the total population.
                        "largest_population": the center longitude/latitude is the centroid of the unit with largest total population. If there is a tie in the maximum population, the mean of all coordinates will be taken.
                    
                    If tuple, this argument should be the coordinates of the desired center in the form (longitude, latitude).
                    
                    If integer, the center will be the centroid of the polygon from data corresponding to the integer interpreted as index. 
                    For example, if `center = 0` the centroid of the first row of data is used as center, if `center = 1` the second row will be used, and so on.

```

I think that the point here is that the function can be generic enough so the user can work with different approaches for the center. Virtually, passing a tuple (second option for the function), the user can fit with any given center (this could be the centroid of the Central Business District if the user knows where it is).

Regarding the LEHD idea, it is pretty cool! However, this is only available only for the US. Therefore, I think we can open an issue only for this task.

I think the multiple cores in the same spatial context should be addressed in a different index of segregation that would deal with that somehow... I never saw one and would be interested if someone finds it!


I really wanted to add @weikang9009  as a reviewer of this PR, but she doesn't appear in the options even when I search for her GitHub username (if someone could add her as a possible reviewer (or allow I add her), I appreciate that):
![image](https://user-images.githubusercontent.com/22593188/57713159-c2c47300-7626-11e9-93b2-ba6b31ef02c1.png)


ps.: This PR was originally created in another that was closed, because I wanted to "clean" my master branch.

knaaptime commented:
I gave the pysal dev team r/w access to this repo so we can request reviews from the larger team (@sjsrey let me know if you'd prefer to handle this differently)

weikang9009 commented on segregation/spatial_indexes.py:
```diff
@@ -1886,8 +1903,33 @@ def _absolute_centralization(data, group_pop_var, total_pop_var):
+            center_lon = c_lons[np.where(t == t.max())].mean()
+            center_lat = c_lats[np.where(t == t.max())].mean()
+
+    if isinstance(center, tuple):
```
`tuple` can be generalized to include `list` and `array` as well? as long as there are two numbers. Sth like this: 
```python
if np.array(center).shape==(2,):
```
Just a suggestion

renanxcortes replied:
Good catch! I resolved this in https://github.com/pysal/segregation/pull/25/commits/3ae9657ae192624657a51fbf1e3ffc820ea5bb35

Also, the print errors are very informative if the user eventually passes an unexpected input.

renanxcortes commented:
> I think for functions, there are no [`Attributes`](https://github.com/pysal/segregation/pull/25/files#diff-0178029925a2d0efbe99487b839b8f91R1861). We should use `Returns`?
> I also think it would be nice to return the center so that users can check where it is if they want.

Yes, returning the centers would be useful! I did this now in https://github.com/pysal/segregation/pull/25/commits/b217baae7cfefc2aa023139eb41814e710eb83d7

I also fixed the 'Attribute' for the internal function and left them only for the classes in https://github.com/pysal/segregation/pull/25/commits/bb73edd5db1cce9144774f8c0bffc562e0fd201b

In addition, I removed some extra lines you pointed out to me last week here: https://github.com/pysal/segregation/pull/25/commits/ec6c670b3c07e25a71edfa4b1f9e36c84108e735

renanxcortes commented:
Thank you for the comments, @weikang9009 !
